### PR TITLE
Remove semantic release and use release-drafter instead

### PR DESCRIPTION
**Pull request by FranceBe**
Widgets could not be released correctly anymore because it used `semantic-release`.
semantic-release did : 
- edit the package.json with the new version
- edit the CHANGELOG.md
- create a tag
- create a commit with the changes
- push the commit into the master branch
- push the commit
- publish the version

The problem in this approach is that the master branch is now a restricted branch. No on can push a commit directly into it, not even semantic-release from the CI.
We needed to re-think the publish process to avoid being blocked for those reason.

What have been done in this PR is : 
- Remove semantic-release dependencies
- Add a workflow with pr-labeler and release-drafter to create a draft release every time a PR is merged
- Add a workflow to publish the package when the draft release is published : 
    - The updated changelog will now be accessible directly into the repository [release section](https://github.com/alma/widgets/releases) (readme has been updated accordingly).
    - The CHANGELOG.md file has been deleted since it won't be used anymore
    - A specific label has been created to flag a PR as "skip changelog" if we don't want to share a specific change  in the changelog
    - The build & the updated package.json will be packaged and published into NPM
    - The version in the package.json will no be updated directly into the repository, but it does not matter since we only need it updated in the published package
- dist/ folder has been removed from the repository & gitignored since we only need it as a package to publish

There is no trying to push on the master branch anymore from this workflow.

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - 'type: security'
+  - title: '⚙️ Internal Changes'
+    labels:
+      - 'type: chore'
+      - 'type: ci'
+      - 'type: perf'
+      - 'type: refactor'
+      - 'type: test'
+change-template: '- $TITLE (#$NUMBER • @$AUTHOR)'
+change-title-escapes: '\<*_&' # You can add # and @ to disable mentions, and add ` to disable code blocks.
```
Comment (noe-charmet) [inline .github/release-drafter.yml:29]: You will have to add the `tag-prefix: v` option
As your tags are currently all using that prefix.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Use Node.js 16
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - run: npm ci
+      - run: |
+          echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}' > .npmrc
+          cat .npmrc
+        # We do not create a commit with the update package version
+        # Therefore the package.*.json in the repository will always have out-of-sync versions
```
Comment (noe-charmet) [inline .github/workflows/release.yml:23]: Not blocking for this PR, but we need to use the new standardised way of referencing the token in npmrc

Comment (FranceBe) [reply .github/workflows/release.yml:23]: oh sure, I was not aware a new standardised way. Of course it'll be better

**Review by noe-charmet — APPROVED** [review]
Let's try it out

File: .github/workflows/release.yml (hunk 2 above)
Comment (noe-charmet) [reply .github/workflows/release.yml:23]: We decided it together 🤦

Comment (FranceBe) [reply .github/workflows/release.yml:23]: oh THAT new standardised way xD

**Review by hyahiaoui — APPROVED** [review]

**Pull request merged**